src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts

```diff
@@ -118,6 +118,13 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
     return !this.pageValidationService.isPageValid(this.currentPage, this.editForm);
   }
 
+  toPreviousPage() {
+    let pageFormFields = this.formValueService.filterCurrentPageFields(this.currentPage.case_fields, this.editForm.value);
+    let caseEventData: CaseEventData = this.formValueService.sanitise(pageFormFields) as CaseEventData;
+    this.updateFormData(caseEventData as CaseEventData);
+    this.previous();
```

Comment:
It seems that there are some steps that should be added sanitiseDynamicLists and caseEventData.event_data = this.editForm.value.data; others I will check it

Reply:
DONE